### ISSUE TITLE
Investigate and fix potential security issue

### ISSUE THREAD

**Issue by colintheshots** [mozilla-mobile/fenix#4686]
Fix this security issue:
https://bugzilla.mozilla.org/show_bug.cgi?id=1572886

**Comment by abodea** [comment]
Verified as fixed on the latest [Fenix build v1.3-rc.3](https://github.com/mozilla-mobile/fenix/releases/tag/v1.3.0-rc.3) with Samsung Galaxy S10+(Android 9). 
Note that I tried the steps from the description of [1572886](https://bugzilla.mozilla.org/show_bug.cgi?id=1572886). 
```css
1. Open page in Firefox Preview (e.g. google for "test")
2. Switch to Twitter App (or any other app that uses Custom Tab)
3. Open page in Twitter custom tab
4. Click 3 dots -> "Open in Firefox Preview"
5. Notice URL bar still has url from the page at (*)
```
Also, I tried with more opened tabs to reproduce it but I couldn't. 
Everything worked as expected the `URL` was changed every time.
I opened more than 15 different links from external apps and I couldn't find any regressions.

@colintheshots I hope this verification is a valid one, but I would like a confirmation from you that this is ok.
Based on my comment I will close this issue.
